tensorboard/webapp/app_routing/route_config.ts

```diff
@@ -198,19 +212,56 @@ class RouteConfigMatcher {
     }
     return pathParts;
   }
+}
 
-  /**
-   * In case of parameter projection failure (cannot create pathname), it throws
-   * a RangeError.
-   */
-  matchByParams(params: RouteParams): PositiveMatch {
-    const pathParts = this.reprojectPathByParams(this.pathFragments, params);
+class ConcreteRouteConfigMatcher extends RouteConfigMatcher {
+  constructor(readonly definition: ConcreteRouteDef) {
```

Comment:
Can we drop the entire constructor?

Reply:
Notice that this constructor _requires_ `ConcreteRouteDef` as opposed to any `RouteDef`. I don't think I can drop it.